flowlayout/src/sharedTest/kotlin/com/google/accompanist/flowlayout/FlowLayoutTest.kt

```diff
@@ -1097,6 +1100,300 @@ class FlowTest : LayoutTest() {
+                height = size * maxLines + spacing * (maxLines - 1)
+            ),
+            flowSize.value
+        )
+        for (i in 0 until requestedNumberOfSquares) {
+            val squareSize = childSize[i].value
+            val squarePosition = childPosition[i].value
+            if (i < expectedNumberOfSquares) {
+                assertEquals(
+                    IntSize(width = size, height = size),
+                    squareSize
+                )
+
+                assertEquals(
+                    Offset(
+                        x = (size * (i % squaresInRow)).toFloat(),
+                        y = ((size + spacing) * (i / squaresInRow)).toFloat()
+                    ),
+                    squarePosition
+                )
+            } else {
+                assertNull(squareSize)
+                assertNull(squarePosition)
+            }
+        }
+    }
+
+    @Test
+    fun testFlowRow_withMaxLines_negative() {
+        val requestedNumberOfSquares = 15
+        val size = 48
+        val maxLines = -1
+        val spacing = 32
+        val flowWidth = 256
+
+        val flowSize = Ref<IntSize>()
+        val childSize = Array(requestedNumberOfSquares) { Ref<IntSize>() }
+        val childPosition = Array(requestedNumberOfSquares) { Ref<Offset>() }
+        val positionedLatch = CountDownLatch(1)
+
+        rule.setContent {
+            Box {
+                ConstrainedBox(
+                    constraints = DpConstraints(
+                        maxWidth = with(LocalDensity.current) { flowWidth.toDp() }
+                    ),
+                    modifier = Modifier.onGloballyPositioned { coordinates: LayoutCoordinates ->
+                        flowSize.value = coordinates.size
+                        positionedLatch.countDown()
+                    }
+                ) {
+                    FlowRow(
+                        maxLines = maxLines,
+                        crossAxisSpacing = with(LocalDensity.current) { spacing.toDp() }
+                    ) {
+                        for (i in 0 until requestedNumberOfSquares) {
+                            Container(
+                                width = with(LocalDensity.current) { size.toDp() },
+                                height = with(LocalDensity.current) { size.toDp() },
+                                modifier = Modifier.saveLayoutInfo(
+                                    childSize[i],
+                                    childPosition[i],
+                                    positionedLatch
+                                )
+                            ) {
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        assertTrue(positionedLatch.await(1, TimeUnit.SECONDS))
+
+        assertEquals(
+            IntSize(width = 0, height = 0),
+            flowSize.value
+        )
+        for (i in 0 until requestedNumberOfSquares) {
```

Comment:
While this behavior is semantically correct, is it really what we want? I'd argue that `maxLines < 0` is an illegal argument and would expect a runtime crash.

Reply:
Fixed, thank you